### DISCUSSION TITLE
Feature Request: Bind to Unix-domain socket

### DISCUSSION THREAD

**Discussion by dgcampea** [jcorporation/myMPD#713] [Ideas]
Unix sockets have less overhead and should be easier (and safer?) to use for reverse-proxied mympd setups.
A hypothetical [nginx](https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass) setup using this would be instead:
```
location /mympd/ {
  proxy_pass http://unix:/run/mympd.socket:;
}
```

**Comment by jcorporation** [comment]
myMPD relies on mongoose for the webserver implementation and mongoose does not support local domain sockets.

You can ask for support of domain sockets in the mongoose repository. If this functionality is added to mongoose, myMPD will also support it.

**Comment by jcorporation** [comment]
https://github.com/cesanta/mongoose/issues/1498

**Comment by dgcampea** [comment]
Revisiting this topic, I've noticed that mongoose has a function called [`mg_wrapfd()`](https://mongoose.ws/documentation/#mg_wrapfd). It should be possible to manually setup a Unix Domain socket and then pass it on for mongoose to use.

**Reply by jcorporation** [reply]
That’s interesting. I have no experience with unix domain sockets, but it it’s worth an attempt.
